dist/jquery.interactive-world-map.coffee

```diff
@@ -0,0 +1,118 @@
+      if $.isPlainObject(settings.countries)  # Check if comes by js object or data-* parameter
+        dataCountries = settings.countries
+      else
+        dataCountries = jQuery.parseJSON(settings.countries)
+      @createPatterns()
+      @prependModalContainer()
+
+
+    createPatterns: ->
+      patterns =
+        highlighted: createSVGPattern(settings.ids.dots.highlighted, settings.colors.highlighted)
+        general:     createSVGPattern(settings.ids.dots.general, settings.colors.general)
+        hover:       createSVGPattern(settings.ids.dots.hover, settings.colors.hover)
+    prependModalContainer: ->
+      $(@element).prepend '<div class="' + settings.classes.popup.modal + '"><div class="' + settings.classes.popup.container + '"><div class="' + settings.classes.popup.containerImage + '" ><div class="' + settings.classes.popup.image + ' flag flag-icon-background  " alt=""></div></div><div><div class="' + settings.classes.popup.title + '"></div><div class="' + settings.classes.popup.containerWebsites + '"></div></div></a></p></div></div>'
+
+  $.fn[pluginName] = (options) ->
+    @each ->
+      new InteractiveWorldMap(this, options)
+      #Call the constructor
+      $.each dataCountries, (index, value) ->
+        countriesFill[index] = 'url(#dotsHighlighted)'
+
+      $(this).vectorMap
+        map:                'world_mill_en'
+        backgroundColor:    'white'
+        dotsGeneralId:       settings.ids.dots.general
+        heightGeneralDots:   '5'
+        widthGeneralDots:    '5'
+        regionStyle:
+          initial: fill:     'url(#' + settings.ids.dots.general + ')'
+          hover: 'fill-opacity': 1
+        series:
+          regions: [ {
+            values: countriesFill
+            attribute: 'fill'
+            hover:
+              'fill-opacity': 0.8
+              cursor: 'pointer'
+        } ]
+
+        onRegionOver: (e, code) ->
+          if dataCountries[code]
+            data = new Array
+            data[code] = 'url(#' + settings.ids.dots.hover + ' )';
+            mapObject = $(this).parent().vectorMap('get', 'mapObject');
+            mapObject.series.regions[0].setValues data;
+
+        onRegionOut: (e, code) ->
+          if dataCountries[code]
+            data = new Array
+            data[code] = 'url(#' + settings.ids.dots.highlighted + ')';
+            mapObject = $(this).parent().vectorMap('get', 'mapObject');
+            mapObject.series.regions[0].setValues data;
+
+        onRegionClick: (e, code) ->
+          if !dataCountries[code]
+            return
+          optionsPopup =
+            code: code
+            margins: defaults.popup.margins
+            country: dataCountries[code]
+            element: $(this)
+            classes: defaults.classes.popup
+          new  $.fn.interactiveWorldMapCountryPopup optionsPopup
+
+      #Apply the svg patterns to the DOM
+      $('defs').html patterns.general + patterns.highlighted + patterns.hover
```

Comment:
Is the `defs` selector really correct to make multiple map use possible? Wouldn't that select all `def`s on the enitre page if I use multiple map instances?